lib/theme_check/checks/pagination_size.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+module ThemeCheck
+  class PaginationSize < LiquidCheck
+    severity :suggestion
+    categories :performance
+    doc docs_url(__FILE__)
+
+    attr_reader :max_size
+
+    def initialize(max_size: 50)
+      @max_size = max_size
+    end
+
+    def on_document(_node)
+      @paginations = {}
+      @schema_settings = {}
+    end
+
+    def on_paginate(node)
+      size = node.value.page_size
+      unless @paginations.key?(size)
+        @paginations[size] = []
+      end
+      @paginations[size].push(node)
+    end
+
+    def on_schema(node)
+      schema = JSON.parse(node.value.nodelist.join)
+
+      if (settings = schema["settings"])
+        @schema_settings = settings
+      end
+    rescue JSON::ParserError
+      # Ignored, handled in ValidSchema.
+    end
+
+    def after_document(_node)
+      @paginations.each_pair do |size, nodes|
+        numerical_size = if size.is_a?(Numeric)
```

Comment:
sorry if I'm missing some assumptions here or if this is handled in some way before this check, but `is_a?(Numeric)` here actually would return true for `-10000`, `-1.5`, `0`, `3.33333` -> wouldn't we want to warn the user about using anything other than a positive integer:
```
if size.is_a?(Integer) &&. size.positive?
```

Reply:
Good point

Reply:
Resolved with later if check.